manifests/smlbench.toml

```diff
@@ -1,18 +1,27 @@
 name = "smlbench"
-# hashicorp/go-getter URLs, so in the future we can support fetching test plans
-# from GitHub.
+# hashicorp/go-getter URLs, so in the future we can support fetching test plans from GitHub.
 source_path = "file:${TESTGROUND_REPO}/plans/smlbench"
 
-[build_strategies."docker:ipfs"]
+# Fetches latest go-ipfs
+[build_strategies."docker:go"]
 enabled = true
 go_version = "1.13"
 module_path = "github.com/ipfs/testground/plans/dht"
+exec_pkg = "."
+go_ipfs_version = "0.4.22"
+
+# Builds latest go-ipfs
+[build_strategies."docker:ipfs"]
+enabled = true
+go_version = "1.13"
```

Comment:
❓ I haven't been able to figure out what is the dark magic that makes these variables become available to the Docker file in an uppercase form

Reply:
For example, for go_version, in pkg/build/docker_go.go, you can see it is configured to look for "go_version" in the TOML file here:

https://github.com/ipfs/testground/blob/5284895c6c4bca284dc61e18c52a9f1094b711bf/pkg/build/docker_go.go#L34

I wasn't really familiar with the third element in the struct there ... those are an advanced feature called "tags" which let you add extra information that is available via reflection.

https://medium.com/golangspec/tags-in-golang-3e5db0b8ef3e

The tags are used by the TOML parser (executed by pkg/engine/engine.go) to map fields:

https://github.com/ipfs/testground/blob/5284895c6c4bca284dc61e18c52a9f1094b711bf/pkg/engine/engine.go#L245

It's an advanced feature of the TOML parser, documented here:

https://godoc.org/github.com/BurntSushi/toml#hdr-Key_mapping

Later, you can see how how it builds the GO_VERSION environment setting for Docker:

https://github.com/ipfs/testground/blob/5284895c6c4bca284dc61e18c52a9f1094b711bf/pkg/build/docker_go.go#L155